quickwit-serve/src/rest.rs

```diff
@@ -58,28 +59,43 @@ pub(crate) async fn start_rest_server(
         ))
         .or(ingest_handler(quickwit_services.push_api_service.clone()))
         .or(tail_handler(quickwit_services.push_api_service.clone()))
+        .or(index_management_handlers(
+            quickwit_services.index_service.clone(),
+        ));
+    let rest_routes = api_v1_root_url
+        .and(api_v1_routes)
+        .or(liveness_check_handler())
         .or(metrics_service)
         .with(request_counter)
         .recover(recover_fn);
+
     info!("Searcher ready to accept requests at http://{rest_addr}/");
     warp::serve(rest_routes).run(rest_addr).await;
     Ok(())
 }
 
 /// This function returns a formated error based on the given rejection reason.
 pub async fn recover_fn(rejection: Rejection) -> Result<impl Reply, Rejection> {
-    // TODO handle more errors.
-    match rejection.find::<serde_qs::Error>() {
-        Some(err) => {
-            let err_msg = err.to_string();
-            Ok(Format::PrettyJson.make_reply_for_err(FormatError {
-                code: ServiceErrorCode::BadRequest,
-                error: err_msg,
-            }))
-        }
-        None => Ok(Format::PrettyJson.make_reply_for_err(FormatError {
+    if rejection.is_not_found() {
```

Comment:
Nitpick: I think the early return style would be better here given we might have more custom error to handle.

Reply:
Actually, I hesitated to make early returns but here I think readability is quite the same with or without.